mesonbuild/interpreter.py

```diff
@@ -473,6 +474,27 @@ def partial_dependency_method(self, args, kwargs):
     def variable_method(self, args, kwargs):
         return self.held_object.get_variable(**kwargs)
 
+    @FeatureNew('dep.is_system', '0.52.0')
+    @noPosargs
+    @permittedKwargs({})
+    def is_system_method(self, args, kwargs):
+        return self.held_object.get_is_system()
+
+    @FeatureNew('dep.as_system', '0.52.0')
+    @permittedKwargs({})
+    def as_system_method(self, args, kwargs):
+        args = listify(args)
+        new_is_system = True
+        if len(args) > 1:
+            raise InterpreterException('as_system takes only one optional value')
+        if len(args) == 1:
+            if not isinstance(args[0], bool):
+                raise InterpreterException('as_system takes only a boolean value')
+            new_is_system = args[0]
+        new_dep = copy.deepcopy(self.held_object)
+        new_dep.is_system = new_is_system
```

Comment:
This should not be here, it should be a method in the dependency class. Something like `create_system_dep`.

Reply:
Done